tensor2tensor/rl/rl_trainer_lib.py

```diff
@@ -71,22 +73,33 @@ def define_train(hparams, environment_spec, event_dir):
       # Some environments reset environments automatically, when reached done
       # state. For them we shall record only every second episode.
       d = 2 if env_lambda().metadata.get("semantics.autoreset") else 1
-      eval_env_lambda = lambda: gym.wrappers.Monitor(  # pylint: disable=g-long-lambda
+      monitor_env_lambda = lambda: gym.wrappers.Monitor(  # pylint: disable=g-long-lambda
           env_lambda(), event_dir, video_callable=lambda i: i % d == 0)
       eval_env_lambda = (
```

Comment:
It is important to have the right hand side lambda (`monitor_env_lambda`) to have a different name than the left hand side. Otherwise it becomes an infinite recursion, see:

```
x = lambda: 1
x = lambda: x()
x()
```